src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -218,6 +224,28 @@ export class CourseExercisesComponent implements OnInit, OnChanges, OnDestroy, A
         return !this.activeFilters.has(ExerciseFilter.UNRELEASED) || (exercise as QuizExercise)?.visibleToStudents;
     }
 
+    /**
+     * Method is called when enter key is pressed on search input
+     * @param event The event of the enter key action
+     */
+    onSearch(event: any) {
+        this.searchFailed = false;
+        if (this.course) {
+            this.isSearching = true;
+            this.courseExerciseService.findAllExercisesForCourseBySearchTerm(this.courseId, event.target.value).subscribe({
+                next: (httpResponse: HttpResponse<Exercise[]>) => {
+                    this.course!.exercises = httpResponse.body ? httpResponse.body : undefined;
+                    this.applyFiltersAndOrder();
+                    this.isSearching = false;
```

Comment:
Use finally operator instead of copying this in next and error
https://www.learnrxjs.io/learn-rxjs/operators/utility/finalize

Reply:
I removed isSearching because searching is client side now.